[master] Update docs for APIM on k8s (#3940)
wso2/docs-apim: en/docs/install-and-setup/install/deploying-api-manager-with-kubernetes-or-openshift-resources.md

    
    - Add the WSO2 Helm chart repository.
        
          ```
           helm repo add wso2 https://helm.wso2.com && helm repo update
          ```

1.  Checkout the Helm Resources for WSO2 API Manager Git repository using `git clone` :

    ```
    git clone https://github.com/wso2/kubernetes-apim.git
    git checkout tags/v4.0.0.1
    ```

2.  Provide the necessary configurations.

    !!! note
        The default product configurations for deployment of WSO2 API Manager are available [here](https://github.com/wso2/kubernetes-apim/tree/v4.0.0.1/simple/am-single/templates) folder. Change the configurations, as necessary.

>     Open the `<HELM_HOME>/am-single/values.yaml` and provide the following values for WSO2 Subscription Configurations.

Reviewer: ```suggestion
    Open the `<HELM_HOME>/am-single/values.yaml` file and provide the following values for WSO2 Subscription Configurations.
```

        If you do not have an active WSO2 subscription, do not change the parameters `wso2.subscription.username` and `wso2.subscription.password`. 

3.   Deploy WSO2 API Manager with Analytics.
    

>     Refer [Register for Analytics](https://apim.docs.wso2.com/en/latest/observe/api-manager-analytics/configure-analytics/register-for-analytics/) to setup the cloud Analytics solution and obtain an on-prem key.

Reviewer: ```suggestion
    Setup the cloud Analytics solution and obtain an on-prem key. For more information, see [Register for Analytics](https://apim.docs.wso2.com/en/latest/observe/api-manager-analytics/configure-analytics/register-for-analytics/).
```

    ```
    helm install --dependency-update --name <RELEASE_NAME> <HELM_HOME>/am-single --namespace <NAMESPACE> --set wso2.choreoAnalytics.enabled=true --set wso2.choreoAnalytics.endpoint=<CHOREO_ANALYTICS_ENDPOINT> --set wso2.choreoAnalytics.onpremKey=<ONPREM_KEY>
    ```

>     You can also provide the above configurations in `<HELM_HOME>/am-single/values.yaml`.

Reviewer: ```suggestion
    You can also provide the above configurations in the `<HELM_HOME>/am-single/values.yaml` file.
```

        ``` java
        NAME                                               HOSTS                                ADDRESS          PORTS      AGE
        wso2am-single-node-am-gateway-ingress               <RELEASE_NAME>-gateway                <EXTERNAL-IP>    80, 443    7m
        wso2am-single-node-am-ingress                       <RELEASE_NAME>-am                     <EXTERNAL-IP>    80, 443    7m
        wso2am-single-node-am-websub-ingress                <RELEASE_NAME>-websub                 <EXTERNAL-IP>    80, 443    7m
        ```

    2.  Add the above hosts as entries in `/etc/hosts` file as follows:
    
          ```
          <EXTERNAL-IP>	<RELEASE_NAME>-am
          <EXTERNAL-IP>	<RELEASE_NAME>-gateway
          <EXTERNAL-IP>	<RELEASE_NAME>-websub
          ```

    3.  Try navigating to `https://<RELEASE_NAME>-am/carbon`, `https://<RELEASE_NAME>-am/publisher` and `https://<RELEASE_NAME>-am/devportal` from your favorite browser.
    
!!! note
    You can read the [README guide](https://github.com/wso2/kubernetes-apim/blob/v4.0.0.1/simple/am-single/README.md) of WSO2 API Manager Git repository for further details on other dependencies and configurations. 

> Refer the following for deployment configurations of [Advanced Deployment Patterns](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/deployment-overview/).

Reviewer: ```suggestion
For more information, see the deployment configurations with regard to the [Advanced Deployment Patterns]({{base_path}}/install-and-setup/setup/deployment-overview/).
```
